com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -382,7 +383,8 @@ public static void NetworkHide(List<NetworkObject> networkObjects, ulong clientI
 
         private void OnDestroy()
         {
-            if (NetworkManager != null && NetworkManager.IsListening && NetworkManager.IsServer == false && IsSpawned)
+            if (NetworkManager != null && NetworkManager.IsListening && NetworkManager.IsServer == false && IsSpawned
```

Comment:
Just wanting to understand the complexity here, we will not Despawn this object if the IsSceneObject is valid and the Value is true... So we are trying the ability to DeSpawn to the lifetime of the IsSceneObject? and does the exception make sense now? because it could be the previous state but also could not be because the IsSceneObject state is not correct.

Reply:
Yeah, I am not 100% happy with the addition of throwing an exception when a "non-server" is destroying things.  The complexity comes into play on the client side during edge case scenarios specific to unloading a scene and object pools.
In-Scene placed NetworkObjects get destroyed when a scene is unloaded, with additive scenes you can load and unload additive scenes multiple times within the same network session, and as such to avoid destroying the stack scope during the unloading of a scene this had to be added.  During an ungraceful exit (i.e. you just decide to exit the application or exit play mode), you will get exceptions for dynamically spawned NetworkObjects on the client if you try to "clean up your object pool" when your spawn generator is being destroyed.

It is a problematic area... I would like to use Application.WantsToQuit to handle these types of edge case scenarios where we can set some NetworkManager state (i.e. NetworkManager.IsApplicationQuitting) that would ignore all of this If statement completely under that condition.

But yeah, it is a problematic area for sure.